test/test-tty.c

```diff
@@ -28,7 +28,7 @@
 #else /*  Unix */
 # include <fcntl.h>
 # include <unistd.h>
-# if defined(__linux__)
+# if defined(__linux__) || (defined(__FreeBSD_kernel__) && defined(__GNUC__))
```

Comment:
`defined(__linux__)` already covers other libc implementations.  That's why we must keep it.